mathics/core/definitions.py

```diff
@@ -725,25 +725,122 @@ def get_history_length(self):
 
 
 def get_tag_position(pattern, name) -> Optional[str]:
+    """
+    Determine the position of a pattern in
+    the definition of the symbol ``name``
+    """
+    blanks = (
+        "System`Blank",
+        "System`BlankSequence",
+        "System`BlankNullSequence",
+    )
+
+    def strip_pattern_name_and_condition(pat):
```

Comment:
Add annotations: 

`strip_pattern_name_and_condition(pat: ExpressionPattern) -> ExpressionPattern`

this will be in PR bark_if_builtin_rule_cannot_be_stored-more